### PR TITLE
Add basic textarea validation and pass valid to helper text

### PR DESCRIPTION
**Pull request by KaneFreeman**
**Type:** feature

The following has been addressed in the PR:

* [ ] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

Add basic validation to the textarea widget, along with an `onValidate` event to match `TextInput`.

**Use case**: checking the validity of a whole form. Currently validation logic for textarea needs to be written each time the widget is used (or a wrapper widget needs to be written).

### CODE REVIEW

File: src/text-area/index.ts [hunk 1]
```diff
@@ -47,6 +53,7 @@ export interface TextareaProperties
 	placeholder?: string;
 	value?: string;
 	onClick?(value: string): void;
+	onValidate?: (valid: boolean | undefined) => void;
```
Comment (tomdye) [inline src/text-area/index.ts]: does this not need `valid?: { valid?: boolean; message?: string } | boolean;` plus logic for displaying it?

Comment (KaneFreeman) [reply src/text-area/index.ts]: Yup, you are right. Will add that.

**Comment by tomdye** [comment]
@KaneFreeman i've picked this up to make the required changes, will push to your branch shortly

File: src/text-area/example/index.ts [hunk 2]
```diff
@@ -5,7 +5,11 @@ import Textarea from '../../text-area/index';
 export default class App extends WidgetBase {
 	private _value1: string | undefined;
 	private _value2: string | undefined;
-	private _invalid: boolean | undefined;
+	private _valid1: undefined | boolean | { valid?: boolean; message?: string };
+	private _customValid: undefined | boolean | { valid?: boolean; message?: string };
+	private _customValue: string | undefined;
+
+	// private _valid1: undefined | { valid: boolean, message?:string };
```
Comment (ericos) [inline src/text-area/example/index.ts]: Remove comment.

**Pull request merged**